src/terraform/commands.go

```diff
@@ -17,11 +18,72 @@ func RunInit(deployConfig types.DeployConfig) error {
 	return err
 }
 
-// RunPlan runs the 'terraform plan' command and points to a secrets file
-func RunPlan(deployConfig types.DeployConfig) error {
-	err := util.RunAndLog(deployConfig.TerraformDir, []string{}, deployConfig.LogChannel, "terraform", "plan", fmt.Sprintf("-var-file=%s", deployConfig.SecretsPath))
+// RunPlan runs the 'terraform plan' command and passes variables in as flags
+func RunPlan(deployConfig types.DeployConfig, secrets types.Secrets) error {
+	command := []string{"terraform", "plan"}
+	vars, err := CompileVarFlags(deployConfig, secrets)
+	if err != nil {
+		return err
+	}
+	command = append(command, vars...)
+	err = util.RunAndLog(deployConfig.TerraformDir, []string{}, deployConfig.LogChannel, command...)
 	if err != nil {
 		return errors.Wrap(err, "'terraform plan' failed")
 	}
 	return err
 }
+
+// CompileVarFlags compiles the variable flags passed into a Terraform command
+func CompileVarFlags(deployConfig types.DeployConfig, secrets types.Secrets) ([]string, error) {
```

Comment:
Thoughts on moving this and its helper functions into its own file. Then can also put the tests for this in its own file.